### DISCUSSION TITLE
How to delete/rename/move files inside helix?

### DISCUSSION THREAD

**Discussion by xtheeq** [helix-editor/helix#11399] [Q&A]
Fuzzy finder makes for searching part, but how to do the delete/rename/move for a file inside helix?

**Comment by David-Else** [comment]
To rename a file:
```
:move, :mv	Move the current buffer and its corresponding file to a different path
```
To delete and move files use terminal commands like `:sh rm filename`.

**Reply by xtheeq** [reply]
I came across [yazi](https://github.com/sxyazi/yazi), which is being the one stop solution for all kinds of file management.

**Reply by David-Else** [reply]
I use [nnn](https://github.com/jarun/nnn) and think it is the best file manager.

**Comment by gokayburuc** [comment]
Basically you can use `:move` function in Helix. 

To Rename: 
- `:mv` file and rename file and enter

To Change Folder: 
-  open your file 
- `:mv` your file like this

```bash 
:mv /newfoldername/mytext.md 

```

To Delete : 
1. You can `:move` file into `/trash` folder. 
2. You can *run-shell-command* and use `rm` in bash.

**Comment by David-Else** [comment]
Things have advanced now with macro key bindings, you can use these shortcuts to create a new file, delete a file and rename a file:

```
[keys.normal.C-space]
n = "@:o <C-r>%<C-w>"
x = "@:sh rm <C-r>%"
r = "@:sh mv <C-r>% <C-r>%"
```
